app/controllers/search_controller.rb

```diff
@@ -0,0 +1,13 @@
+class SearchController < ApplicationController
+  def index
+    @events = []
+    # raw_response = Faraday.get 'http://api.eventful.com/json/events/search?app_key=7J7CGGFM5zgs8RH7&location=80202&date=2017030100-2017033100&sort_order=popularity&page_size=16&page_number=1'
+    # raw_response = Faraday.get "http://api.eventful.com/json/events/search?app_key=7J7CGGFM5zgs8RH7&location=#{params[:q]}&date=2017030100-2017033100&sort_order=popularity&page_size=16&page_number=1"
+    # response = JSON.parse(raw_response.body)
+    # requested_events = response["events"]["event"]
+    requested_events = EventFinderService.new(location: params[:q]).events
+    requested_events.each do |one_event|
```

Comment:
@case-eee Thank you for the comment. Absolutely makes sense. Unfortunately I didn't have time to move it to `EventFinderService`. There is so much more refactor need around the `controller` and `EventFinderService`. :-)